File: DoubanMusic.lbaction/Contents/Scripts/default.php
<?php
require_once('workflows.php');
$w = new Workflows();
$query = urlencode( "{query}" );
$api = "01068bdd0c3168a70313a397249439f5";
$url = "https://api.douban.com/v2/music/search?count=20&apikey=$api&q=$query";
$suggestions = $w->request( $url );
$suggestions = json_decode( $suggestions );
function get_name($i) {
   return $i->name;
}
foreach( $suggestions->musics as $suggest ):
   $w->result( $suggest->id, $suggest->alt, $suggest->title, '表演者: '. implode(",", $suggest->attrs->singer) .' 评分: '. $suggest->rating->average .'/'. $suggest->rating->numRaters .' 标签: '. implode(",", array_map('get_name', $suggest->tags)), '67BE2A80-D1E6-4A47-B82A-E140CBA759A0.png' );
endforeach;

if ( count( $w->results() ) == 0 ):
         $w->result( 'dobanmusic', 'http://music.douban.com/subject_search?search_text='. $query, '糟糕...', '没找到符合条件的音乐, 去豆瓣搜搜看？', 'icon.png', 'yes' );
endif;   

echo $w->toxml();

File: DoubanMusic.lbaction/Contents/Scripts/default.py
#!/usr/bin/env python
# -*- coding:utf-8 -*-
#
# LaunchBar Action Script
#
import sys
import json
import urllib
import urllib2

API = "01068bdd0c3168a70313a397249439f5"
URL = "https://api.douban.com/v2/music/search?count=20&"

items = []
item = {}

keyword = "".join(sys.argv[1:])
if not keyword:
	sys.exit()

query = urllib.urlencode({'apikey': API, 'q': keyword})
api_url = URL + query
res = urllib2.Request(api_url)
response = urllib2.urlopen(res)
html = response.read()
response.close()
if not html:
    sys.exit()

resultList = json.loads(html)

for music in resultList['musics']:
	item = {}
	item['title'] = music['title']
	item['subtitle'] = u'表演者: ' + ",".join(music['attrs']['singer']) + u' 评分: ' + str(music['rating']['average']) + '/' + str(music['rating']['numRaters'])
	item['icon'] = '67BE2A80-D1E6-4A47-B82A-E140CBA759A0.png'
	item['url'] = music['alt']
	item['quickLookURL'] = music['alt']
	items.append(item)

print json.dumps(items)
